tests/cmdline/commands/test_node.py

```diff
@@ -557,39 +561,267 @@ def test_rehash_invalid_entry_point(self, run_cli_command):
+
+                for i in range(n_calcjobs):
+                    calcjob_node = CalcJobNode(computer=aiida_localhost)
+                    calcjob_node.add_incoming(workflow_node, link_type=LinkType.CALL_CALC, link_label='call')
+
+                    workdir = tmp_path / f'calcjob_{uuid.uuid4()}'
+                    workdir.mkdir()
+                    Path(workdir / 'fileA.txt').write_text('test stringA')
+                    self.remote_folders.append(workdir)
+
+                    calcjob_node.set_remote_workdir(str(workdir))
+                    calcjob_node.set_option('output_filename', 'fileA.txt')
+                    calcjob_node.store()
+                    self.calcjob_nodes.append(calcjob_node)
+                    self.calcjob_pks.append(calcjob_node.pk)
+
+                    for _ in range(n_remotes):
+                        remote_node = RemoteData(remote_path=str(workdir), computer=aiida_localhost)
+                        remote_node.base.links.add_incoming(calcjob_node, LinkType.CREATE, link_label='remote_folder')
+                        remote_node.store()
+                        self.remote_nodes.append(remote_node)
+                        self.remote_pks.append(remote_node.pk)
+
+            # this is equal to (n_workflows + n_workflows * n_calcjobs + n_workflows * n_calcjobs * n_remotes)
+            self.total_nodes = len(self.workflow_nodes) + len(self.calcjob_nodes) + len(self.remote_nodes)
+
+        return _setup
+
+    def verify_deletion(self, nodes_deleted=True, folders_deleted=True):
+        """Verify that the nodes and remote folders are deleted or not deleted."""
+
+        if nodes_deleted:
+            for workflow_pk in self.workflow_pks:
+                with pytest.raises(NotExistent):
+                    WorkflowNode.objects.get(pk=workflow_pk)
+
+            for calcjob_pk in self.calcjob_pks:
+                with pytest.raises(NotExistent):
+                    CalcJobNode.objects.get(pk=calcjob_pk)
+
+            for remote_pk in self.remote_pks:
+                with pytest.raises(NotExistent):
+                    RemoteData.objects.get(pk=remote_pk)
+        else:
+            for workflow_pk in self.workflow_pks:
+                WorkflowNode.objects.get(pk=workflow_pk)
+            for calcjob_pk in self.calcjob_pks:
+                CalcJobNode.objects.get(pk=calcjob_pk)
+            for remote_pk in self.remote_pks:
+                RemoteData.objects.get(pk=remote_pk)
+
+        for remote_folder in self.remote_folders:
+            if folders_deleted:
+                assert not remote_folder.exists()
+            else:
+                assert remote_folder.exists()
+
+    def test_setup_node_hierarchy(self, setup_node_hierarchy):
+        """Test the `setup_node_hierarchy` and `verify_deletion` fixtures."""
+        # Guard the guardians
+        setup_node_hierarchy()
+        self.verify_deletion(nodes_deleted=False, folders_deleted=False)
```

Comment:
If no nodes are created, the `for` loop will be skipped, and the `verify_deletion` will still pass, so first check the nodes are indeed created.
```suggestion
        setup_node_hierarchy()
        # make sure the nodes are created
        assert len(self.workflow_nodes) == 1
        assert len(self.calcjob_nodes) == 1
        assert len(self.remote_nodes) == 1
        self.verify_deletion(nodes_deleted=False, folders_deleted=False)
```

Reply:
Very good catch :1st_place_medal:! of my tests :see_no_evil: 

Fixed now!